src/amg/azext_amg/_params.py

```diff
@@ -66,6 +66,12 @@ def load_arguments(self, _):
         c.argument("archive_file", options_list=["-a", "--archive-file"], help="archive to restore Grafana artifacts from")
         c.argument("remap_data_sources", options_list=["-r", "--remap-data-sources"], arg_type=get_three_state_flag(),
                    help="during restoration, update dashboards to reference data sources defined at the destination workspace through name matching")
+        
+    with self.argument_context("grafana migrate") as c:
+        c.argument("source_instance_ip", options_list=["-s", "--src-ip"], help="Grafana instance IP to migrate from")
+        c.argument("source_instance_token", options_list=["-t", "--src-token"], help="Grafana instance token to migrate from")
+        c.argument("dry_run", options_list=["-d", "--dry-run"], arg_type=get_three_state_flag(), help="Preview changes without committing. Takes priority over --overwrite.")
+        c.argument("overwrite", options_list=["--overwrite"], arg_type=get_three_state_flag(), help="Overwrite previous dashboards, library panels, and folders with the same uid with new ones (version replacement)")
```

Comment:
I feel like the overwrite help line can be cleaned up, there's some redundant language

Reply:
How about "Replace existing dashboards, panels, and folders with new ones using the same UID", basically, this flag will allow customers to overwrite previous dashboards with same uid. There is an edge case where there is a title conflict but different uid, it will keep the old uid but then overwrite everything else. How do you think I phrase that weird case?

Reply:
Are you saying if the new dashboards match existing dashboards in title, and you use the --overwrite flag, it will overwrite the dashboard, but keep the existing UID?

Can you clarify for me how conflicts are identified? And then how they are resolved when this --overwrite flag is used and when it is not used?

Reply:
Yup, so there a couple cases (due to quirks of the Grafana HTTP API).

For dashboard specifically, when we send a request to create the dashboard, it will either create it successfully or not. If it doesn't it will return with a status and error code. Depending on the error code, we will pass the message to the user. So Grafana API will check that the Dashboard Title or UID is taken. If it is taken, then it will return if an error.

Now the overwrite flag comes into play when we receive the flag, we can pass it into the request and when we do, Grafana API will automatically overwrite. However, through testing I noticed that:
- If UIDs match, everything will be overwritten (including title, if title is different)
- If Titles match, everything will be overwritten EXCEPT it will keep the old (existing) dashboard UID
- If both UID and title match, then UID and title will be same and the dashboard data will just be overwritten (so new panels, etc).

Reply:
I see. If it overwrites based on UID or Title maybe "Overwrite previous dashboards, library panels, and folders with the same uid or title" would be good? I think we should include it will also overwrite based on title.

Reply:
Okay that sounds good!

Reply:
I also added support for Snapshots & Annotations, but snapshot can have same title and annotations will only overwrite if certain conditions are true.

Reply:
Ok, so my understanding here then is dashboards and folders use the Grafana HTTP API overwrite flags, but library panels and snapshots and annotations use our own defined behavior. 

I don't think it will be feasible to explain the nuance of all of these cases in the help line. We want to most avoid cases where the user doesn't expect something to be overwritten when it is, so I think mentioning both uid and title is a necessity. The inverse case where something is not overwritten that was expected to be is less harmful.

The only other necessary change I think is removing "with new ones." I think this is redundant as it is implied by overwriting,